### PR TITLE
First draft of physics features.

### PR DESCRIPTION
**Pull request by moorepants**
minted has been included for syntax highlighting (make sure to invoke the
included Makefile for the correct complilation flag)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pdf:
```
Comment (asmeurer) [inline Makefile]: You need to add `all: pdf`

**Comment by asmeurer** [comment]
You didn't include minted.

**Comment by moorepants** [comment]
minted should be included in texlive-latex-extras.

File: Makefile (hunk 1 above)
Comment (moorepants) [inline Makefile]: I think make builds the first target by default if you don't have all, but I'll add it.

I need to add that stop on error command back in too.

**Comment by moorepants** [comment]
Oh, maybe the version of ubuntu that is on travis doesn't have minted in the apt repos. Will need to check. It works on 15.10 just fine.

File: domain_specific.tex [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+>>> A = ReferenceFrame('A')
+>>> B = ReferenceFrame('B')
+>>> C = ReferenceFrame('C')
+>>> B.orient(A, 'body', (1, 2, 3), 'xzx')
+>>> C.orient(B, 'axis', (4, B.x))
+>>> v = 1 * A.x + 2 * B.z + 3 * C.y
+>>> v
+A.x + 2*B.z + 3*C.y
+>>> v.express(A)
+(-3*sin(2)*cos(3)*cos(4) + 3*sin(2)*sin(3)*sin(4) + 2*sin(2)*sin(3) + 1)*A.x + (3*(-sin(3)*cos(1)*cos(2) - sin(1)*cos(3))*sin(4) + 3*(-sin(1)*sin(3) + cos(1)*cos(2)*cos(3))*cos(4) - 2*sin(3)*cos(1)*cos(2) - 2*sin(1)*cos(3))*A.y + (2*cos(1)*cos(3) + 3*(sin(3)*cos(1) + sin(1)*cos(2)*cos(3))*cos(4) - 2*sin(1)*sin(3)*cos(2) + 3*(cos(1)*cos(3) - sin(1)*sin(3)*cos(2))*sin(4))*A.z
```
Comment (asmeurer) [inline domain_specific.tex]: You need to wrap this manually.

**Comment by asmeurer** [comment]
Is it good practice to just add the style file to the repo? 

Otherwise, I think maybe you need to add some PPA.

**Comment by asmeurer** [comment]
I get a few bibtex errors when building.

File: domain_specific.tex [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+The vector package provides reference frame, time, and space aware vector and
+dyadic objects that allow for three dimensional operations such as addition,
+subtraction, scalar multiplication, inner and outer products, cross products,
+etc. Both of these objects can be written in very compact notation that make it
+easy to express the vectors and dyadics in terms of multiple reference frames
+with arbitrarily defined relative orientations. The vectors are used to specify
+the positions, velocities, and accelerations of points, orientations, angular
+velocities, and angular accelerations of reference frames, and force and
+torques. The dyadics are essentially reference frame aware 3 x 3 tensors. The
```
Comment (asmeurer) [inline domain_specific.tex]: `$3 \times 3$`

File: domain_specific.tex [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+vector and dyadic objects can be used for any one, two, or three dimensional
```
Comment (asmeurer) [inline domain_specific.tex]: one-, two-, or three-dimensional

File: domain_specific.tex [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+vector algebra and provide a strong framework to build physics and engineering
+tools atop.
```
Comment (asmeurer) [inline domain_specific.tex]: This reads awkward. Maybe "provide a strong framework for building physics and engineering tools."

File: domain_specific.tex [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+\begin{listing}
+  \begin{minted}{pycon}
+>>> A = ReferenceFrame('A')
```
Comment (asmeurer) [inline domain_specific.tex:25]: Add `from sympy.vector import *` (correct?) to the top.

File: domain_specific.tex [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+>>> B = ReferenceFrame('B')
+>>> C = ReferenceFrame('C')
+>>> B.orient(A, 'body', (1, 2, 3), 'xzx')
```
Comment (asmeurer) [inline domain_specific.tex]: You should add a comment or description to the caption of what the orient lines do. It's not immediately obvious to me what this means.

File: domain_specific.tex [hunk 8]
```diff
@@ -0,0 +1,78 @@
+SymPy includes several packages that allow users to solve domain specific
+problems. For example, a comprehensive physics package is included that is
+useful for solving problems in classical mechanics, optics, and quantum
+mechanics along with support for manipuating physical quantities with units.
+
+\subsection{Vector Algebra}
+
+The vector package provides reference frame, time, and space aware vector and
```
Comment (asmeurer) [inline domain_specific.tex]: The \verb|sympy.vector| package.

File: domain_specific.tex [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+are autmated linearization routines for constrained dynamical
+systems based on ~\cite{Peterson2014}.
+
+\subsection{Quantum Mechanics}
+
+The \verb|sympy.physics.quantum| package provides quantum functions, states,
+operators, and computation of standard quantum models.
+
+% TODO : This needs some help from someone that knows something about quantum
+% physics. I wasn't able to understand much from the documentation.
```
Comment (asmeurer) [inline domain_specific.tex:70]: Assuming that we can't find someone who can expand this, what should we do?

Comment (moorepants) [inline domain_specific.tex:70]: Let's just open an issue. Maybe I can convince @certik or @ellisonbg to write a few sentences.

**Comment by moorepants** [comment]
Ok, hopefully everything is fixed and this is mergable.

**Comment by moorepants** [comment]
I don't know what all these entries are the in bib file. Are we even using them? For some reason the url field is being detected in the entries in the bib file and then `\usepackage{url}` is being inserted in the `paper.bbl` file. I've never come across this error before. Can we delete all entries in the bibliography that aren't being used? In my PR none of the entries in the bib file are being used except the 3 that I added.

**Comment by moorepants** [comment]
It works now by removing all the unused bib entries.

**Comment by asmeurer** [comment]
I added a bunch of stuff to the bib file that we might use. Doesn't LaTeX work just fine with extra bib entries?

**Comment by moorepants** [comment]
You have url entries that is causing bibtex to balk. I think we should only have the things that we are going to use in the bib file. They can be added and subtracted as needed.

If you want to get all those entries working with the bibtex command you should do that before committing. I don't really want to go through them all and manually fix them. But maybe deleting all url fields would be a quick fix.

**Comment by asmeurer** [comment]
Should we have a separate bib file then? A lot of those references I grabbed en masse (e.g., I copied the references from Mateusz's thesis), but a lot also I grabbed because I explicitly thought we would use them. So I don't want to just throw them out entirely.

**Comment by moorepants** [comment]
I would say bring them in one by one as we need them so that we know they compile. We may want to use biblatex instead of bitext too, as it handles more types of entries and stuff.

**Comment by ellisonbg** [comment]
Hi all, I would be open to contributing some stuff about the
physics/quantum stuff. Some questions:
- What is the timeline?
- What is the target journal?

Cheers, Brian

On Wed, Apr 13, 2016 at 2:15 PM, Jason K. Moore notifications@github.com
wrote:

> I would say bring them in one by one as we need them so that we know they
> compile. We may want to use biblatex instead of bitext too, as it handles
> more types of entries and stuff.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/sympy/sympy-paper/pull/38#issuecomment-209649879

## 

Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by moorepants** [comment]
@ellisonbg The timeline is pretty tight. Aaron wants a first draft by Monday.

We have yet to select a journal but need to soon, as it will likely affect how we pitch the paper. See issue #7.

I think we only need (at most) a paragraph or two about the quantum stuff. My guess is that the physics sub-package sections will get trimmed once we see the breadth of the paper. So if you could write just a handful of sentences that would give us a starting point.

**Comment by ellisonbg** [comment]
OK I can get some text in for that. Just posted a comment on the other
issue about the journal.

On Wed, Apr 13, 2016 at 3:21 PM, Jason K. Moore notifications@github.com
wrote:

> @ellisonbg https://github.com/ellisonbg The timeline is pretty tight.
> Aaron wants a first draft by Monday.
> 
> We have yet to select a journal but need to soon, as it will likely affect
> how we pitch the paper. See issue #7
> https://github.com/sympy/sympy-paper/issues/7.
> 
> I think we only need (at most) a paragraph or two about the quantum stuff.
> My guess is that the physics sub-package sections will get trimmed once we
> see the breadth of the paper. So if you could write just a handful of
> sentences that would give us a starting point.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/sympy/sympy-paper/pull/38#issuecomment-209672729

## 

Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

File: domain_specific.tex [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+operations such as addition, subtraction, scalar multiplication, inner and
+outer products, cross products, etc. Both of these objects can be written in
+very compact notation that make it easy to express the vectors and dyadics in
+terms of multiple reference frames with arbitrarily defined relative
+orientations. The vectors are used to specify the positions, velocities, and
+accelerations of points, orientations, angular velocities, and angular
+accelerations of reference frames, and force and torques. The dyadics are
+essentially reference frame aware $3 \times 3$ tensors. The vector and dyadic
+objects can be used for any one-, two-, or three-dimensional vector algebra and
+they provide a strong framework for build physics and engineering tools.
```
Comment (isuruf) [inline domain_specific.tex]: build->building

File: domain_specific.tex [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+A.x + 5*sqrt(3)/2*A.y + 5/2*A.z
+  \end{minted}
+  \caption{
+    Python interpreter session showing how a vector is created using the
+    orthogonal unit vectors of three reference frames that are oriented with
+    respect to each other and the result of expressing the vector in the $A$
+    frame.
+    The $B$ frame is oriented with respect to the $A$ frame using Z-X-Z Euler
+    Angles of magnitude $\pi$, $\frac{\pi}{2}$, and $\frac{\pi}{3}$ radians,
+    respecitively whereas the $C$ frame is oriented with respect to the $B$
```
Comment (isuruf) [inline domain_specific.tex]: respectively

File: domain_specific.tex [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  \label{lis:physics-vector}
+\end{listing}
+
+\subsection{Classical Mechanics}
+
+The \verb|physics.mechanics| package utilizes the \verb|physics.vector| package
+to populate time aware particle and rigid body objects to fully describe the
+kinematics and kinetics of a rigid multi-body system. These objects store all
+of the information needed to derive the ordinary differential or differential
+algrebraic equations that govern the motion of the system, i.e. the equations
```
Comment (isuruf) [inline domain_specific.tex]: algebraic

File: domain_specific.tex [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+of motion. These equations of motion abide by Newton's laws of motion and can
+handle any arbitrary kinematical constraints or complex loads. The package
+offers two automated methods for formulating the equations of motion based on
+Lagrangian Dynamics~\cite{Lagrange1811} and Kane's Method~\cite{Kane1985}. Lastly, there
+are autmated linearization routines for constrained dynamical
```
Comment (isuruf) [inline domain_specific.tex]: automated

**Comment by fredrik-johansson** [comment]
It looks all right so far.

**Comment by moorepants** [comment]
Ok, shall I merge this? We just got a request for minted in #52, so this would provide that to others when they merge  master.

**Comment by moorepants** [comment]
Merging. I suppose this will introduce some conflicts with others on the bib file.

**Pull request merged**